programs/rodeos/streamer_plugin.cpp

```diff
@@ -0,0 +1,77 @@
+// copyright defined in LICENSE.txt
+
+#include "streamer_plugin.hpp"
+#include "streams/logger.hpp"
+#include "streams/rabbitmq.hpp"
+#include "streams/stream.hpp"
+
+#include <abieos.hpp>
+#include <eosio/abi.hpp>
+#include <fc/exception/exception.hpp>
+#include <memory>
+
+namespace b1 {
+
+using namespace appbase;
+using namespace std::literals;
+
+struct streamer_plugin_impl {
+   std::vector<std::unique_ptr<stream_handler>> streams;
+};
+
+static abstract_plugin& _streamer_plugin = app().register_plugin<streamer_plugin>();
+
+streamer_plugin::streamer_plugin() : my(std::make_shared<streamer_plugin_impl>()) {}
+
+streamer_plugin::~streamer_plugin() {}
+
+void streamer_plugin::set_program_options(options_description& cli, options_description& cfg) {
+   auto op = cfg.add_options();
+   op("stream-rabbits", bpo::value<std::vector<string>>()->composing(),
+      "RabbitMQ Streams if any; Format: amqp://USER:PASSWORD@ADDRESS:PORT/QUEUE[/ROUTING_KEYS, ...]");
+   op("stream-loggers", bpo::value<std::vector<string>>()->composing(),
+      "Logger Streams if any; Format: [routing_keys, ...]");
+}
+
+void streamer_plugin::plugin_initialize(const variables_map& options) {
+   try {
+      if (options.count("stream-loggers")) {
+         auto loggers = options.at("stream-loggers").as<std::vector<std::string>>();
+         initialize_loggers(my->streams, loggers);
+      }
+
+      if (options.count("stream-rabbits")) {
+         auto rabbits = options.at("stream-rabbits").as<std::vector<std::string>>();
+         initialize_rabbits(app().get_io_service(), my->streams, rabbits);
+      }
+
+      ilog("initialized streams: ${streams}", ("streams", my->streams.size()));
+   }
+   FC_LOG_AND_RETHROW()
+}
+
+void streamer_plugin::plugin_startup() {
+   cloner_plugin* cloner = app().find_plugin<cloner_plugin>();
```

Comment:
Since `streamer_plugin` requires on `cloner_plugin` add an assert:
  `EOS_ASSERT( cloner, plugin_config_exception, "cloner_plugin not found" );`

Reply:
Do we not feel `APPBASE_PLUGIN_REQUIRES((cloner_plugin))` is sufficient?

Reply:
The assert is more in case someone removes that in my thought. We do assert similarly for other plugins.